pkg/commands/compute/init.go

```diff
@@ -196,25 +211,104 @@ func (c *InitCommand) Exec(in io.Reader, out io.Writer) (err error) {
 		c.cloneFrom = from
 	}
 
-	// We only want to fetch a remote package if c.cloneFrom has been set.
-	// This can happen in two ways:
+	// There are three situations in which we might fetch something
+	// here. We might fetch a template if:
 	//
-	// 1. --from flag is set
+	// 1. --from flag is set to a template repository, or
 	// 2. user selects starter kit when prompted
 	//
+	// Or we fetch an existing, deployed package if
+	//
+	// 3. --from flag is set to a serviceID
+	//
 	// We don't fetch if the user has indicated their language of choice is
 	// "other" because this means they intend on handling the compilation of code
 	// that isn't natively supported by the platform.
 	if c.cloneFrom != "" {
-		err = c.FetchPackageTemplate(branch, tag, file.Archives, spinner, out)
-		if err != nil {
-			c.Globals.ErrLog.AddWithContext(err, map[string]any{
-				"From":      from,
-				"Branch":    branch,
-				"Tag":       tag,
-				"Directory": c.dir,
+		if !isExistingService {
+			err = c.FetchPackageTemplate(branch, tag, file.Archives, spinner, out)
+			if err != nil {
+				c.Globals.ErrLog.AddWithContext(err, map[string]any{
+					"From":      from,
+					"Branch":    branch,
+					"Tag":       tag,
+					"Directory": c.dir,
+				})
+				return err
+			}
+		} else {
+			profileName, _, _ := c.Globals.Profile()
+
+			var service *fastly.Service
+			var pack *fastly.Package
+			err = spinner.Process("Fetching service details", func(_ *text.SpinnerWrapper) error {
+				service, err = c.Globals.APIClient.GetService(&fastly.GetServiceInput{
+					ServiceID: c.cloneFrom,
+				})
+				if err != nil {
+					c.Globals.ErrLog.AddWithContext(err, map[string]any{
+						"From":      c.cloneFrom,
+						"Directory": c.dir,
+					})
+					if hErr, ok := err.(*fastly.HTTPError); ok && hErr.IsNotFound() {
+						return fmt.Errorf("the service %s could not be found for profile %s", c.cloneFrom, profileName)
+					}
+					return err
+				}
+
+				if service.Type == nil || *service.Type != "wasm" {
+					return fmt.Errorf("service %s is not a compute service (type is %s)", c.cloneFrom, *service.Type)
```

Comment:
I think this needs to be capitalized as it's a product name (best to check throughout the PR in case any other references need updating)...

```suggestion
					return fmt.Errorf("service %s is not a Compute service (type is %s)", c.cloneFrom, *service.Type)
```